### ISSUE TITLE
Unable to get intro_skipper working

### ISSUE THREAD

**Issue by Ritter99** [Casvt/Plex-scripts#9]
First off, thanks for all the work here. I have been enjoying your scripts.

As the title says, I cannot get the intro_skipper.py to work. I have been attempting to do a little debugging of my own and found that neither `Chapter` nor `Marker` are in `media_output[data['PlaySessionStateNotification'][0]['ratingKey']]['MediaContainer']['Metadata'][0].keys()`. 

My stream does get recognized as 'playing' and `media_output[data['PlaySessionStateNotification'][0]['ratingKey']] = json.loads(requests.get('http://' + plex_ip + ':' + str(plex_port) + data['PlaySessionStateNotification'][0]['key'], params={'X-Plex-Token': plex_api_token}, headers={'Accept': 'application/json'}).text)` is assigning data to `media_output` variable.

Is this something that Plex recently changed and broke the script? Or am I just goofing something up?

**Comment by Casvt** [comment]
Could you share the value of media_output please (as you report that it is indeed assigning data to it)? After the declaration of the variable, you could just insert a `print(media_output)` line in there to get it's value.

**Comment by Ritter99** [comment]
Absolutely! Here is the output of print(media_output):

`{
    '17396': {
        'MediaContainer': {
            'size': 1,
            'allowSync': True,
            'identifier': 'com.plexapp.plugins.library',
            'librarySectionID': 2,
            'librarySectionTitle': 'TVShows',
            'librarySectionUUID': 'e4603660-597b-4cc3-bec4-2cb2a997cf88',
            'mediaTagPrefix': '/system/bundle/media/flags/',
            'mediaTagVersion': 1643704412,
            'Metadata': [
                {
                    'ratingKey': '17396',
                    'key': '/library/metadata/17396',
                    'parentRatingKey': '17395',
                    'grandparentRatingKey': '17394',
                    'guid': 'plex: //episode/5d9c186cef619b00204b7f73',
                    'parentGuid': 'plex: //season/602e70ebfdd281002ce05479',
                    'grandparentGuid': 'plex: //show/5d9c08ccba2e21001f192e95',
                    'type': 'episode',
                    'title': 'Secrets',
                    'grandparentKey': '/library/metadata/17394',
                    'parentKey': '/library/metadata/17395',
                    'librarySectionTitle': 'TVShows',
                    'librarySectionID': 2,
                    'librarySectionKey': '/library/sections/2',
                    'grandparentTitle': 'Dark',
                    'parentTitle': 'Season1',
                    'originalTitle': 'Geheimnisse',
                    'contentRating': 'TV-MA',
                    'summary': "In 2019, a local boy's disappearance stokes fear in the residents of Winden, a small German town with a strange and tragic history.",
                    'index': 1,
                    'parentIndex': 1,
                    'audienceRating': 7.6,
                    'viewOffset': 868000,
                    'lastViewedAt': 1645500815,
                    'thumb': '/library/metadata/17396/thumb/1636127511',
                    'art': '/library/metadata/17394/art/1636127510',
                    'parentThumb': '/library/metadata/17395/thumb/1630769925',
                    'grandparentThumb': '/library/metadata/17394/thumb/1636127510',
                    'grandparentArt': '/library/metadata/17394/art/1636127510',
                    'grandparentTheme': '/library/metadata/17394/theme/1636127510',
                    'duration': 3090326,
                    'originallyAvailableAt': '2017-12-01',
                    'addedAt': 1611692502,
                    'updatedAt': 1636127511,
                    'audienceRatingImage': 'themoviedb: //image.rating',
                    'Media': [
                        {
                            'id': 37394,
                            'duration': 3090326,
                            'bitrate': 1939,
                            'width': 1920,
                            'height': 960,
                            'aspectRatio': 1.85,
                            'audioChannels': 6,
                            'audioCodec': 'aac',
                            'videoCodec': 'hevc',
                            'videoResolution': '1080',
                            'container': 'mkv',
                            'videoFrameRate': '24p',
                            'audioProfile': 'he-aac',
                            'videoProfile': 'main10',
                            'Part': [
                                {
                                    'id': 37400,
                                    'key': '/library/parts/37400/1611692502/file.mkv',
                                    'duration': 3090326,
                                    'file': '/data/media/tv_shows/TV-14/DARK/Season1/DARK-S01E01-SecretsWEBRip-1080p.mkv',
                                    'size': 749840598,
                                    'audioProfile': 'he-aac',
                                    'container': 'mkv',
                                    'videoProfile': 'main10',
                                    'Stream': [
                                        {
                                            'id': 80599,
                                            'streamType': 1,
                                            'default': True,
                                            'codec': 'hevc',
                                            'index': 0,
                                            'bitrate': 1522,
                                            'bitDepth': 10,
                                            'chromaSubsampling': '4: 2: 0',
                                            'codedHeight': 960,
                                            'codedWidth': 1920,
                                            'colorRange': 'tv',
                                            'frameRate': 24.0,
                                            'height': 960,
                                            'level': 120,
                                            'profile': 'main10',
                                            'refFrames': 1,
                                            'width': 1920,
                                            'displayTitle': '1080p(HEVCMain10)',
                                            'extendedDisplayTitle': '1080p(HEVCMain10)'
                                        },
                                        {
                                            'id': 80600,
                                            'streamType': 2,
                                            'default': True,
                                            'codec': 'aac',
                                            'index': 1,
                                            'channels': 6,
                                            'bitrate': 210,
                                            'language': 'Deutsch',
                                            'languageTag': 'de',
                                            'languageCode': 'deu',
                                            'profile': 'he-aac',
                                            'samplingRate': 48000,
                                            'displayTitle': 'Deutsch(HE-AAC5.1)',
                                            'extendedDisplayTitle': 'Deutsch(HE-AAC5.1)'
                                        },
                                        {
                                            'id': 80601,
                                            'streamType': 2,
                                            'selected': True,
                                            'codec': 'aac',
                                            'index': 2,
                                            'channels': 6,
                                            'bitrate': 207,
                                            'language': 'English',
                                            'languageTag': 'en',
                                            'languageCode': 'eng',
                                            'profile': 'he-aac',
                                            'samplingRate': 48000,
                                            'displayTitle': 'English(HE-AAC5.1)',
                                            'extendedDisplayTitle': 'English(HE-AAC5.1)'
                                        },
                                        {
                                            'id': 80602,
                                            'streamType': 3,
                                            'default': True,
                                            'codec': 'srt',
                                            'index': 3,
                                            'bitrate': 0,
                                            'language': 'English',
                                            'languageTag': 'en',
                                            'languageCode': 'eng',
                                            'displayTitle': 'English(SRT)',
                                            'extendedDisplayTitle': 'English(SRT)'
                                        },
                                        {
                                            'id': 80603,
                                            'streamType': 3,
                                            'codec': 'srt',
                                            'index': 4,
                                            'bitrate': 0,
                                            'language': 'Nederlands',
                                            'languageTag': 'nl',
                                            'languageCode': 'nld',
                                            'displayTitle': 'Nederlands(SRT)',
                                            'extendedDisplayTitle': 'Nederlands(SRT)'
                                        },
                                        {
                                            'id': 80604,
                                            'streamType': 3,
                                            'codec': 'srt',
                                            'index': 5,
                                            'bitrate': 0,
                                            'language': 'Français',
                                            'languageTag': 'fr',
                                            'languageCode': 'fra',
                                            'displayTitle': 'Français(SRT)',
                                            'extendedDisplayTitle': 'Français(SRT)'
                                        },
                                        {
                                            'id': 80605,
                                            'streamType': 3,
                                            'codec': 'srt',
                                            'index': 6,
                                            'bitrate': 0,
                                            'language': 'Deutsch',
                                            'languageTag': 'de',
                                            'languageCode': 'deu',
                                            'displayTitle': 'Deutsch(SRT)',
                                            'extendedDisplayTitle': 'Deutsch(SRT)'
                                        },
                                        {
                                            'id': 80606,
                                            'streamType': 3,
                                            'codec': 'srt',
                                            'index': 7,
                                            'bitrate': 0,
                                            'language': 'Español',
                                            'languageTag': 'es',
                                            'languageCode': 'spa',
                                            'displayTitle': 'Español(SRT)',
                                            'extendedDisplayTitle': 'Español(SRT)'
                                        },
                                        {
                                            'id': 86047,
                                            'key': '/library/streams/86047',
                                            'streamType': 3,
                                            'codec': 'srt',
                                            'language': 'English',
                                            'languageTag': 'en',
                                            'languageCode': 'eng',
                                            'format': 'srt',
                                            'displayTitle': 'English(SRTExternal)',
                                            'extendedDisplayTitle': 'English(SRTExternal)'
                                        }
                                    ]
                                }
                            ]
                        }
                    ],
                    'Director': [
                        {
                            'id': 61192,
                            'filter': 'director=61192',
                            'tag': 'BaranboOdar'
                        }
                    ],
                    'Writer': [
                        {
                            'id': 61193,
                            'filter': 'writer=61193',
                            'tag': 'JantjeFriese'
                        }
                    ],
                    'Guid': [
                        {
                            'id': 'imdb: //tt6305578'
                        },
                        {
                            'id': 'tmdb: //1279454'
                        },
                        {
                            'id': 'tvdb: //6327940'
                        }
                    ],
                    'Role': [
                        {
                            'id': 61178,
                            'filter': 'actor=61178',
                            'tag': 'DaanLennardLiebrenz',
                            'role': 'MikkelNielsen',
                            'thumb': 'https: //metadata-static.plex.tv/people/5d776aaa23d5a3001f5076d6.jpg'
                        },
                        {
                            'id': 61179,
                            'filter': 'actor=61179',
                            'tag': 'AngelaWinkler',
                            'role': 'InesKahnwald',
                            'thumb': 'https: //metadata-static.plex.tv/people/5d776827961905001eb9145a.jpg'
                        },
                        {
                            'id': 52760,
                            'filter': 'actor=52760',
                            'tag': 'OliverMasucci',
                            'role': 'UlrichNielsen',
                            'thumb': 'https: //metadata-static.plex.tv/d/people/dbb44e4174e5f73c57e9f05c50dc7c6e.jpg'
                        },
                        {
                            'id': 61180,
                            'filter': 'actor=61180',
                            'tag': 'JennipherAntoni',
                            'role': 'UllaObendorf',
                            'thumb': 'https: //metadata-static.plex.tv/5/people/587949693804dfb65674ad31e3add74d.jpg'
                        },
                        {
                            'id': 61181,
                            'filter': 'actor=61181',
                            'tag': 'PaulRadom',
                            'role': 'ErikObendorf'
                        },
                        {
                            'id': 61182,
                            'filter': 'actor=61182',
                            'tag': 'NilsBrunkhorst',
                            'role': 'Teacher'
                        },
                        {
                            'id': 61183,
                            'filter': 'actor=61183',
                            'tag': 'TomJahn',
                            'role': 'JürgenObendorf',
                            'thumb': 'https: //metadata-static.plex.tv/b/people/b9fcd957132a0987a440a149a444f6b4.jpg'
                        },
                        {
                            'id': 61184,
                            'filter': 'actor=61184',
                            'tag': 'MiekeSchymura',
                            'role': 'PoliceOfficerJankowski',
                            'thumb': 'https: //metadata-static.plex.tv/people/5d776e86fb0d55001f5c4ac1.jpg'
                        },
                        {
                            'id': 196208,
                            'filter': 'actor=196208',
                            'tag': 'PeterBenedict',
                            'role': 'AleksanderTiedemann',
                            'thumb': 'https: //metadata-static.plex.tv/0/people/0de949bb7e97f5d215c9c9940a3ec6c5.jpg'
                        },
                        {
                            'id': 61185,
                            'filter': 'actor=61185',
                            'tag': 'MoritzJahn',
                            'role': 'MagnusNielsen',
                            'thumb': 'https: //metadata-static.plex.tv/people/5d7768c1308bca0020331e8b.jpg'
                        },
                        {
                            'id': 61186,
                            'filter': 'actor=61186',
                            'tag': 'GinaStiebitz',
                            'role': 'FranziskaDoppler',
                            'thumb': 'https: //metadata-static.plex.tv/people/5d77689ffb0d55001f515b04.jpg'
                        },
                        {
                            'id': 61187,
                            'filter': 'actor=61187',
                            'tag': 'TatjaSeibt',
                            'role': 'JanaNielsen',
                            'thumb': 'https: //metadata-static.plex.tv/people/5d77682f2e80df001ebdd96d.jpg'
                        },
                        {
                            'id': 61188,
                            'filter': 'actor=61188',
                            'tag': 'DeborahKaufmann',
                            'role': 'ReginaTiedemann',
                            'thumb': 'https: //metadata-static.plex.tv/f/people/f4ae944f69ad928abbd869a150401544.jpg'
                        },
                        {
                            'id': 61189,
                            'filter': 'actor=61189',
                            'tag': 'HermannBeyer',
                            'role': 'HelgeDoppler',
                            'thumb': 'https: //metadata-static.plex.tv/people/5d77682654c0f0001f301d79.jpg'
                        },
                        {
                            'id': 61190,
                            'filter': 'actor=61190',
                            'tag': 'SebastianRudolph',
                            'role': 'MichaelKahnwald',
                            'thumb': 'https: //metadata-static.plex.tv/people/5d7768266f4521001ea98d71.jpg'
                        },
                        {
                            'id': 61191,
                            'filter': 'actor=61191',
                            'tag': 'LeopoldHornung',
                            'role': 'TorbenWöller',
                            'thumb': 'https: //metadata-static.plex.tv/people/5d77683185719b001f3a2b08.jpg'
                        }
                    ]
                }
            ]
        }
    }
}`

**Comment by Casvt** [comment]
Okay can you do something for me? I think I found the problem. Do the following exactly as I say please:

1. Open the file that you currently have on your computer (the one that you added the `print` line to)
2. Go to the line that you mentioned here (I believe that it is line 41): 

> My stream does get recognized as 'playing' and media_output[data['PlaySessionStateNotification'][0]['ratingKey']] = json.loads(requests.get('http://' + plex_ip + ':' + str(plex_port) + data['PlaySessionStateNotification'][0]['key'], params={'X-Plex-Token': plex_api_token}, headers={'Accept': 'application/json'}).text) is assigning data to media_output variable.
3. Change the following
`params={'X-Plex-Token': plex_api_token}`
to
`params={'X-Plex-Token': plex_api_token, 'includeChapters': 1, 'includeMarkers': 1}`
Take notice of the single quotes (`'`), lack there of around the 1's and the usage of capital letters. Just copy it exacty please is what I'm trying to say.
4. Keep the `print(media_output)` that there probably already was.
5. Share the output of media_output again like you did before.

So basically you need to do exactly the same thing but with the params thing changed.
Also please remove your previous comment containing the output. I got it and it isn't needed anymore and you might want to delete it now for privacy reasons (nothing to find in there but you never know...).

**Comment by Casvt** [comment]
I've upgraded the script significantly and added the fix above. Redownload the script from here please and try again.

**Comment by Ritter99** [comment]
It does appear to be working now. Thank you very much!

If you don't mind a recommendation: It would be nice if we could store things like `plex_ip`, `plex_port` and `plex_api_token` in a `.env` file so that we can simply do a git pull on this repo to get updated without filling all our information back in.
